custom/icds/tasks/sms.py

```diff
@@ -72,7 +72,9 @@ def send_custom_sms_report(start_date, end_date, email):
     subject = _('Monthly SMS report')
     recipients = [email]
     filename = call_command('get_icds_sms_usage', 'icds-cas', str(start_date), str(end_date))
+    report_tracker = CustomSMSReportTracker()
     try:
+        report_tracker.add_report_to_list(str(start_date), str(end_date))
```

Comment:
Should not it be added in the view itself? This way it won't reflect until the task has begun.

Reply:
How long does that delay is?
I kept it this way because with this approach registering a report and removing it are happening in same function which I thought would more readable.

Reply:
> How long does that delay is?

the delay is to the point when the task is started, which can go from seconds to few hours. As a user, i would submit a request and then expect to see it in the list on reload if not instant. So we would need to registed it when we accept a request.

Reply:
Ohh! I have not imagined that lag could be in hours. 😮 
I will add it to the view then 👍